### ISSUE TITLE
Does it support isometric tilemap?

### ISSUE THREAD

**Issue by kuyum** [RafaelBarbosatec/bonfire#86]
Thanks for the great work. I am trying dev an isometric game. Does it support isometric tilemap?

**Comment by RafaelBarbosatec** [comment]
Hi @kuyum! Thank's for contact !
We do not have support for an isometric tilemap yet.
It would be a very interesting feature. Even the new versions of Tiled have a very nice support for isometric maps.
If you want to help us implement this it would be very interesting.

**Comment by GyHUN95** [comment]
It's arrived in flame 1.4.0, https://docs.flame-engine.org/1.4.0/flame/components.html?highlight=isometric#isometrictilemapcomponent

**Comment by RafaelBarbosatec** [comment]
Yes, it's already possible render isometric maps. But to give you a good experience with this kind of game. I will need adds the third dimension: x,y,z. And for this I have to create many modifications to give support for all behaviors and components offered by Bonfire.

**Comment by 4mitabh** [comment]
I dont think, we'll need to change any of the data structures, or need a Z dimension.

The common way to implement isometric maps in games uses only x,y coordinates, and most of the game engine remains exactly the same. The cartesian to isometric transformation are required only at two places:
1. When rending tiles on the screen.
2. When reading user input. The up/down/left/right need to be interpreted according to the tilted map displayed on the screen.

Here are two introductory videos on YouTube that talk about this.

https://www.youtube.com/watch?v=04oQ2jOUjkU

https://www.youtube.com/watch?v=KvSjJ-kdGio

**Comment by RafaelBarbosatec** [comment]
Hi @4mitabh !

To render it's OK. But to component’s movements if there are height of different tiles. How a can handle with this without z dimension? Thinking better, i guess it’s possible. I don’t know. I'll study and search more about this.
Thanks so much for share this videos!
I think that I'll start the support of this without thinking in z dimension.

**Comment by 4mitabh** [comment]
I see what you meant.

Yeah, we'll have to take care of z-order of all the rendered objects. In case of isometric objects, the trick is to simply adjust the z-order of the rendered objects according to the y coordinates. 

We already do this correctly in the current maps. The knight can go around the barrel, both behind and in front of it.
So, it should just work in the isometric maps too.

**Comment by invennicoofficial** [comment]
Any plan on this?

**Comment by RafaelBarbosatec** [comment]
I'm focused in release the bonfire 3.0 and the new Bonfire documentation. After this I will work in the isometric games.

**Comment by gochev** [comment]
I would love to make an isometric game with bonfire, bonfire is super awesome for beginners.


Btw instead of Z usually most games are made just using x and y even StarCraft :) and for the Z there is just a top, ground and underground

**Comment by BrownySprite47** [comment]
Is there a plan to add support for isometric maps? Is there any information on the timing? Very necessary.

**Comment by anna-galochkina** [comment]
In general, I managed to achieve isometry for tiles (this fix is ​​enough for my project). Only the tile should be square, for example 128x128. But here you can play with the height of the tiles and choose for yourself. And then everything turns out in accordance with the map in the Tiled application.

fix for TiledWorldBuilder._addTile

```
  var x = _getX(count, tileLayer.width?.toInt() ?? 1);
  var y = _getY(count, tileLayer.width?.toInt() ?? 1);

  if (y % 2 != 0) {
      x = x + 0.5;
  }
 
 _layers.last.tiles.add(
      Tile(
          x: x,
          y: y / 4,
          ...
      ),
  );

```
